packages/components/src/ConditionalFilter/GroupFilter.tsx

```diff
@@ -317,7 +326,7 @@ const GroupFilter: React.FunctionComponent<GroupFilterProps> = ({
                 {menuItems.length > 0 && <MenuGroup>{renderItems(menuItems as GroupFilterItem[])}</MenuGroup>}
                 {groupMenuItems.map((group: GroupFilterItem | Group, groupKey: number) => (
                   <MenuGroup
-                    label={(group as Group).groupSelectable && typeof group.label === 'string' ? group.label : undefined}
+                    label={!(group as Group).groupSelectable && typeof group.label === 'string' ? group.label : undefined}
```

Comment:
https://github.com/RedHatInsights/frontend-components/blob/123a089577ca093cb9eb39decf23ec28f41578bf/packages/components/src/ConditionalFilter/GroupFilter.js#L190

Original code for reference. I was thinking of using something similar to what was used before, but the condition was even more complex. `(group as Group).groupSelectable ? undefined : (group.label === 'string' ? group.label : undefined)`. Although checking the docs, we can render anything, not only string.

Reply:
Oh, patternfly has incorrect types https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/Menu/MenuGroup.tsx#L11 I'll open an issue and PR on their side to allow React.Node elements as well.

Reply:
Oh yeah, there are a few of these. Can you send the PF PR for reference? We might want to do some typecasting anyway so we are not blocked until next stable release.

Reply:
I've opened an issue so far https://github.com/patternfly/patternfly-react/issues/8209 while coding the PR I realised even cheekier approach. What if we'd want to send a component directly to the header? Currently working on writing tests etc.